store/src/java/com/zimbra/cs/service/FileUploadServlet.java

```diff
@@ -766,6 +803,38 @@ List<Upload> handlePlainUpload(HttpServletRequest req, HttpServletResponse resp,
         items.add(fi);
         Upload up = new Upload(acct.getId(), fi, filename);
 
+        if (!acct.isFeatureFileTypeUploadRestrictionsEnabled()) {
+            if (filename.endsWith(".har")) {
+                File file = ((DiskFileItem) fi).getStoreLocation();
+                try {
+                    Tika tika = new Tika();
+                    String mimeType = tika.detect(file);
+                    if (mimeType != null) {
+                        up.contentType = mimeType;
+                    }
+                } catch (IOException e) {
+                    mLog.warn("Failed to detect file content type");
+                }
+            }
+            final String finalMimeType = up.contentType;
+            String contentTypeBlacklist = LC.zimbra_file_content_type_blacklist.value();
+            List<String> blacklist = new ArrayList<String>();
+            if (!StringUtil.isNullOrEmpty(contentTypeBlacklist)) {
+                blacklist.addAll(Arrays.asList(contentTypeBlacklist.trim().split(",")));
+            }
+            if (blacklist.stream().anyMatch((blacklistedContentType) -> {
+                Pattern p = Pattern.compile(blacklistedContentType);
+                Matcher m = p.matcher(finalMimeType);
+                return m.find();
+            })) {
+                mLog.debug("handlePlainUpload(): deleting %s", fi);
+                fi.delete();
+                mLog.info("File content type is blacklisted : %s" + finalMimeType);
```

Comment:
Replace + with ,

Reply:
it's a revert and hence we should not fix review comments, this code is anyways going to be removed again when we implement this feature properly